# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineCacheIRCompiler.cpp
# Commit: 3f8a0fa88566
# Full Hash: 3f8a0fa8856668b5c0e40f91cdd5b95bb5ea9ea2
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-07-07 09:47:47
# Regressor Bug: 1646378
# File Overlap Count: 2
# Description:
#   Bug 1646378: Get stubspace from ICScript in AttachBaselineCacheIRStub r=jandem
#   
#   My current plan for stub allocation is as follows. A JitScript contains a fallbackStubSpace, which is used for all of its own stubs. When we decide to do trial inlining for a JitScript, we create an InliningRoot, which is owned by the JitScript. The InliningRoot contains an array of unique pointers to inlined ICScripts and has a single fallbackStubSpace which is shared by all those ICScripts.
#   
#   I wrote this code when I was thinking that we would throw the InliningRoot away after Warp compilation. If we intend to keep it around indefinitely, it might be better to reuse the stubspace in the root JitScript, in which case we don't need this patch.
# ==============================================================================

diff -r f657de7a7c64 -r 3f8a0fa88566 js/src/jit/BaselineCacheIRCompiler.cpp
--- a/js/src/jit/BaselineCacheIRCompiler.cpp	Thu Jul 02 22:58:05 2020 +0000
+++ b/js/src/jit/BaselineCacheIRCompiler.cpp	Thu Jul 02 22:30:45 2020 +0000
@@ -2086,7 +2086,7 @@
 
 ICStub* js::jit::AttachBaselineCacheIRStub(
     JSContext* cx, const CacheIRWriter& writer, CacheKind kind,
-    BaselineCacheIRStubKind stubKind, JSScript* outerScript,
+    BaselineCacheIRStubKind stubKind, JSScript* outerScript, ICScript* icScript,
     ICFallbackStub* stub, bool* attached) {
   // We shouldn't GC or report OOM (or any other exception) here.
   AutoAssertNoPendingException aanpe(cx);
@@ -2229,8 +2229,8 @@
 
   size_t bytesNeeded = stubInfo->stubDataOffset() + stubInfo->stubDataSize();
 
-  ICStubSpace* stubSpace =
-      ICStubCompiler::StubSpaceForStub(stubInfo->makesGCCalls(), outerScript);
+  ICStubSpace* stubSpace = ICStubCompiler::StubSpaceForStub(
+      stubInfo->makesGCCalls(), outerScript, icScript);
   void* newStubMem = stubSpace->alloc(bytesNeeded);
   if (!newStubMem) {
     return nullptr;